src/store/modules/chatMessages.js

```diff
@@ -7,10 +7,10 @@ const state = {
 // getters
 const getters = {
   // Get virtual concierge items
-  getChatMessagesOrderedByDate(state) {
-    if (state.chatMessages.length !== 0) {
-      // [...state.chatMessages[0].messages] makes a copy of the state array. Getters cannot mutate the state directly.
-      const conversation = [...state.chatMessages[0].messages].sort((a, b) => {
+  getChatMessageByIdOrderedByMessagesDate: (state, getters) => (id) => {
+    let cm = { ...getters.getChatMessageById(id) };
+    if (cm.messages.length !== 0) {
```

Comment:
If cm is not what we expect, cm.messages will crash.

We need to check before the `if (cm.messages.......)` for the content of cm. 

Maybe add a `if (!cm) return {}`

or if cm returns as an empty object if nothing is found - (see comment below) - we could add this check:
`Object.keys(cm).length === 0`

Reply:
I put in a check to see if cm is undefined. If it isn't it will escape with the existing object which is still undefined